utils/xpath_query.py

```diff
@@ -0,0 +1,133 @@
+from seismograph.ext import selenium
+from seismograph.ext.selenium.exceptions import PollingTimeoutExceeded
+from seismograph.ext.selenium.query import QueryObject, QueryResult, logger
+from seismograph.utils.common import waiting_for
+from selenium.common.exceptions import NoSuchElementException, WebDriverException
+
+
+class XPathQueryObject(QueryObject):
+    def __init__(self, xpath, **selector):
+        super(XPathQueryObject, self).__init__(selenium.query.ANY, **selector)
+        self.xpath = xpath
+
+    def __call__(self, proxy):
+        return make_result(proxy, self.tag)(self.xpath)
+
+
+class XPathQueryResult(QueryResult):
+    def __init__(self, proxy, css):
+        super(XPathQueryResult, self).__init__(proxy, css)
+        self.__we = None
+
+        self.__css = css
+        self.__proxy = proxy
+
+    def __repr__(self):
+        return u'{}({}): {}'.format(
+            self.__class__.__name__, self.__css, repr(self.__proxy),
+        )
+
+    def __getattr__(self, item):
+        if not self.__we:
+            self.first()
+        return getattr(self.__we, item)
+
+    @property
+    def exist(self):
+        """
+        Check exist first element of query
+
+        :rtype: bool
+        """
+        try:
+            el = execute(self.__proxy, self.__css, disable_polling=True)
+
+            if el:
+                return True
+            return False
+        except WebDriverException:
+            return False
+
+    def wait(self, timeout=None, delay=None):
+        """
+        Wait for first element of query while timeout doesn't exceeded
+
+        :param timeout: time for wait in seconds
+        """
+        return waiting_for(
+            lambda: self.exist,
+            exc_cls=PollingTimeoutExceeded,
+            delay=delay or self.__proxy.config.POLLING_DELAY,
+            timeout=timeout or self.__proxy.config.POLLING_TIMEOUT,
+            message='Could not wait web element by xpath "{}"'.format(self.__css),
+        )
+
+    def get(self, index):
+        """
+        Get element of query by index
+
+        :param index: index of element
+        """
+        try:
+            self.__we = execute(self.__proxy, self.__css, list_result=True)[index]
+            return self.__we
+        except IndexError:
+            raise NoSuchElementException(
+                'Result does not have element by index "{}". XPath query: "{}".'.format(
+                    index, self.__css,
+                ),
+            )
+
+    def first(self):
+        """
+        Get first element of query
+        """
+        self.__we = execute(self.__proxy, self.__css)
+        return self.__we
+
+    def all(self):
```

Comment:
Вот теперь понятно, при попытке найти элемент через css селектор с contains у текста, geckodriver падает с ошибкой `WebDriverException: Message: Error communicating with the remote browser. It may have died.` Похоже на баг в драйвере. 
У меня заработало с 
```
        selenium.query(
            selenium.query.A,
            href='/payment/service_22',
        )
```